pkg/icingadb/driver.go

```diff
@@ -55,8 +55,8 @@ func (c RetryConnector) Connect(ctx context.Context) (driver.Conn, error) {
 		shouldRetry,
 		backoff.NewExponentialWithJitter(time.Millisecond*128, time.Minute*1),
```

Comment:
With the nesting of the `retry.WithBackoff()` within those in other places that execute SQL queries and these backoff settings, some strange effects can be observed: If you stop your database for something like 4m30s, you may see some attempts recovering and others - quite some time after that - failing fatally because those exceeded the 5 minutes. I believe this happens to the maximal backoff of a minute given here, i.e. there may be a minute without an attempt and after that, both the inner and outer 5 minute timeout are exceeded.

Also, if we're adding `retry.WithBackoff()`, do we even still need it here? Just for fun, I removed all the retry from this function (so that it just does `Connect()` and `initConn()` with no loops or anything around it) and it seemed to work more reliably, only the logging became very annoying with queries and stacktraces within a single line.

Reply:
The following changes address your considerations:

`HA` now uses a `Timer` instead of a `Ticker` that is reset properly even if expired (and not yet drained): 81085c04a42a67527559a611f8129af26afbfb13.

If the timeout expires during the sleep phase between attempts one final retry attempt will be made: 3a3baafc86064f398e98972f71ff9bd709e7149f.

> only the logging became very annoying with queries and stacktraces within a single line

I have something in my stash which fixes this. `zap` adds the `errorVerbose` field if the error implements `fmt.Formatter` which is true for all our wrapped errors. Simple solution would be to replace `zap.Error()` calls with a custom implementation that returns a "silent" error, e.g.

```go
package logging

...

type stackTracer interface {
	StackTrace() errors.StackTrace
}

type errNoStackTrace struct {
	e error
}

func (e errNoStackTrace) Error() string {
	return e.e.Error()
}

func Error(e error) zap.Field {
	if _, ok := e.(stackTracer); ok {
		return zap.Error(errNoStackTrace{e})
	}

	return zap.Error(e)
}
```

Reply:
> > only the logging became very annoying with queries and stacktraces within a single line
> 
> I have something in my stash which fixes this. `zap` adds the `errorVerbose` field if the error implements `fmt.Formatter` which is true for all our wrapped errors. Simple solution would be to replace `zap.Error()` calls with a custom implementation that returns a "silent" error, e.g.

However, the PR just keeps the nested retrying for now, thereby avoiding this issue.